hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/timeline/versioning/v2/TimelineArchiverV2.java

```diff
@@ -117,6 +118,11 @@ public int archiveIfRequired(HoodieEngineContext context, boolean acquireLock) t
         deleteArchivedActions(instantsToArchive, context);
         // triggers compaction and cleaning only after archiving action
         this.timelineWriter.compactAndClean(context);
+        List<HoodieInstant> archivedInstants = instantsToArchive.stream()
```

Comment:
same as above, the logic looks redundant because at here the archiving for Hoodie already finished.